### ISSUE TITLE
run_pytest_check: 300 s timeout is hardcoded and unreachable

### ISSUE THREAD

**Issue by MarcusJellinghaus** [MarcusJellinghaus/mcp-code-checker#214]
# `run_pytest_check`: 300 s timeout is hardcoded and unreachable

## Problem

`run_pytest_check` cannot be used on a project whose suite takes longer than 300 s.
The timeout is not exposed as a tool parameter, a server flag, or a config value.

`checker_tools/pytest_tool.py` calls `check_code_with_pytest(...)` without
`timeout_seconds`, so the default applies:

```python
# code_checker_pytest/runners.py:91 and :473
timeout_seconds: int = 300,
```

On overrun the caller gets a single opaque line (`checker_tools/__init__.py:68`):

```
Error running pytest: Process timed out after 300 seconds
```

It does not say the limit is unconfigurable, nor suggest scoping the run — so the
caller's only apparent move is to retry, which cannot help.

## Impact

The caller can never reach a green result; there is no knob to turn. In an
automated agent loop this is fatal: the agent is told "all checks must pass",
pytest can only ever return an error, and the workflow stalls.

Measured on `mcp_coder` (4879 tests) on an idle developer machine. Note this
project sets `-n auto` in `pyproject.toml` `addopts`, so **every** run below is
already parallel — the variable is which markers run:

| Invocation | Result |
|---|---|
| unit tests only (integration markers excluded) | **256 s** — passes, 15 % under the limit |
| integration markers included | **timed out at 300 s** (reproduced) |
| default `run_pytest_check()` — no marker filter | times out |

Even the fastest sensible invocation leaves ~44 s of headroom on an idle machine;
a loaded CI agent erases that. And a project with a genuinely slow suite has no
recourse at all.

Compare: `--refactoring-timeout` is already configurable (`main.py:97`); pytest —
by far the longest-running tool — is not.

## Fix

Resolve the timeout from four sources, first match wins:

1. **`timeout_seconds` tool argument** — new `Optional[int] = None` on
   `run_pytest_check`, passed through to `check_code_with_pytest`.
2. **`pyproject.toml`** — e.g. `[tool.mcp-tools-py] pytest_timeout = 900`.
   The right value is a property of the project's suite, so it belongs with the
   project and should travel with the repo rather than be re-set in every
   launcher. `utils/project_config.py` already reads pyproject.toml
   (`get_target_directories`), and every checker tool documents "Auto-detected
   from pyproject.toml" — this extends an existing pattern rather than inventing
   one.
3. **`--pytest-timeout` server flag** — mirrors `--refactoring-timeout`
   (`main.py:97`). Ranked below pyproject because in practice it is set once in a
   shared launcher template and acts as a cross-project default rather than a
   deliberate per-project choice.
4. **300 s default** — unchanged.

Also: report a timeout distinctly instead of folding it into the generic
`Error running pytest:` string. The caller should be able to tell "the suite is
too slow for the limit" from "pytest crashed", and the message should name the
effective limit, where it came from, and how to raise it.

Applying the same resolution to the other checkers' timeouts (mypy and pylint are
both hardcoded at 120 s) would be consistent, though neither is currently a
blocker.

## Note

`README.md` states "Parallel test execution is enabled by default using
pytest-xdist (`-n auto`)". `runners.py` does not add `-n auto`; it only applies
when the project's own `addopts` supplies it, as `mcp_coder` does. Either
implement it or correct the README.
